app/helpers/view_helper.rb

```diff
@@ -76,6 +76,8 @@ def govuk_button_to(name, options = {}, html_options = {})
       draggable: false,
     }.merge(html_options)
 
+    return button_to(name, options, html_options) { yield } if block_given?
```

Comment:
I think we might need something similar to https://github.com/rails/rails/blob/b9ca94caea2ca6a6cc09abaffaad67b447134079/actionview/lib/action_view/helpers/url_helper.rb#L300 or https://github.com/DFE-Digital/apply-for-teacher-training/pull/4078/commits/61f33f20eff4e3e9ecb8800df4af8189683f5f8d#diff-20b0835bb2cec2d019f0fc54cbaa1457a740c573dba553c3333d7521de832092L52 where we need to check if a block is given, and then set the variables accordingly. We could also allow a block to be passed into the method. Might look something like this:

```
def govuk_button_to(name, options = {}, html_options = {}, &block)
  if block_given?
    html_options = options
    options = name
  end

  html_options = {
      class: prepend_css_class('govuk-button', html_options[:class]),
      role: 'button',
      data: { module: 'govuk-button' },
      draggable: false,
    }.merge(html_options)

  return button_to(options, html_options) { yield } if block_given?
  button_to(name, options, html_options)
end
```